Add tooling for building the jetpack forms dashboard

ice9js opened the pull request:
## Proposed changes:

This patch adds the necessary tooling and setup to continue work on the jetpack forms dashboard refresh, originally started in #27071.

It's currently set up to have a separate submenu entry under 'Jetpack' which will make it easy to work on it and chances are that's where it will eventually end up anyway.  
That said, we'll likely initially allow users to access the old dashboard anyway until we've reached feature parity but setting that up goes beyond the scope of this PR.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pcsBup-Fk-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

You will need to enable the `automattic/jetpack-forms` package as well as the new dashboard before testing, by making sure these two filters are added before the contact form module is loaded:

```php
add_filter( 'jetpack_contact_form_use_package', '__return_true' );
add_filter( 'jetpack_forms_dashboard_enable', '__return_true' );
```

Make sure the package is installed with `jetpack install packages/forms && jetpack install plugins/jetpack`, and built the JS bundle with `jetpack build packages/forms`. You can also test the `watch` functionality with `jetpack watch packages/forms`.

- Go to WP Admin.
- There should be a new 'Jetpack Forms' submenu item inside 'Jetpack'.
- When you proceed clicking on it, you should see the new dashboard. Not it's not much for now ;)

ice9js committed:
Add forms dashboard app skeleton

ice9js committed:
Add .eslintrc.js

ice9js committed:
Add tooling for building the forms dashboard app

ice9js committed:
Hook up the new dashboard in WP_Admin

ice9js committed:
changelog

jeherve committed:
Merge remote-tracking branch 'origin/trunk' into add/forms-dashboard-setup

jeherve committed:
Do not ship lint config file in production

jeherve committed:
webpack files are not needed in production buiuld

jeherve committed:
Add docblock for new filter

jeherve committed:
Fix textdomain in webpack config

jeherve committed:
[not verified] Merge remote-tracking branch 'origin/trunk' into add/forms-dashboard-setup

jeherve committed:
Build both bundles with the jetpack cli

jeherve committed:
Ship dist bundle files in production

jeherve commented on projects/packages/forms/.gitattributes:
```diff
@@ -6,8 +6,9 @@ package.json      export-ignore
-/src/**/*.rtl.css production-include
+/src/**/*.min.css      production-include
+/src/**/*.rtl.css      production-include
+/src/dashboard/dist/** production-include
```
I'm thinking it may be easier if that bundle lived at the root of the package, instead of in a subdirectory? What do you think?

jcheringer replied:
Kuba and I had this conversation yesterday and agreed to have it at the root since https://github.com/Automattic/jetpack/pull/28630 will also need a `dist` folder.

jeherve replied:
Alright. I've updated it in a5147d31bbb4f3a17651c804bb24bee43282f920

jeherve committed:
Update watch format.

This will allow us to build all things Forms with the Jetpack CLI:

`jetpack build packages/forms`
`jetpack watch packages/forms`

jeherve committed:
Change location of the dashboard build

See https://github.com/Automattic/jetpack/pull/28689#discussion_r1094466938

jeherve committed:
Simplify script enqueuing

jeherve committed:
Only load the dashboard js in the page where we need it.

jeherve committed:
Do not ship raw files in production build

jeherve committed:
Add basic babel config

jeherve committed:
Clean and validate build

jeherve reviewed (approved):
I believe this should be good to go now. I've made a few changes, feel free to go through them.

I also updated this PR's description following my changes to the packages' `package.json` scripts. I believe this should make it easier for all of us to build and develop the package that way, relying on the Jetpack CLI.

jeherve commented on projects/packages/forms/src/dashboard/class-dashboard.php:
```diff
@@ -0,0 +1,79 @@
+		Admin_Menu::add_menu(
+			__( 'Jetpack Forms', 'jetpack-forms' ),
+			_x( 'Jetpack Forms', 'product name shown in menu', 'jetpack-forms' ),
+			'read',
```
Coming back to this, maybe we should change this to `edit_pages`, to match the old Feedback menu?
